service/storage_proxy.cc

```diff
@@ -1742,7 +1723,7 @@ class storage_proxy::cancellable_write_handlers_list : public bi::list<abstract_
         _live_iterators.push_back(itp);
     }
     void unregister_live_iterator(iterator* itp) {
-        _live_iterators.erase(boost::remove(_live_iterators, itp), _live_iterators.end());
+        _live_iterators.erase(std::ranges::remove(_live_iterators, itp).begin(), _live_iterators.end());
```

Comment:
Wouldn't it be more canonical to do:

```
const auto [begin, end] =  std::ranges::remove(_live_iterators, itp);
_live_iterators.erase(begin, end);
```

Reply:
Maybe, I tried to minimize changes.

We could/should do an improvements pass later. I saw a lot of improvement opportunities.